src/api/routes/auth.py

```
@@ -0,0 +1,29 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from src.database import models
+from src.database.connection import get_db
+from src.utils.models import UserCreate, UserResponse
+from src.utils.helpers import hash_password, verify_password
+
+router = APIRouter()
+
+@router.post("/register", response_model=UserResponse)
+def register_user(user: UserCreate, db: Session = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)